src/llmcompressor/transformers/calibration/oneshot.py

```diff
@@ -0,0 +1,244 @@
+
+        # Clean up the CompressionSession before exit if requested
+        if self.recipe_args.clear_sparse_session:
+            self.lifecycle.reset()
+
+    def apply_recipe_modifiers(self, calibration_dataloader: Optional[DataLoader]):
+        self.lifecycle.initialize(
+            model=self.model,
+            recipe=self.recipe_args.recipe,
+            recipe_args=self.recipe_args.recipe_args,
+            calib_data=calibration_dataloader,
+            start=-1,  # oneshot specific arg
+            copy_data=False,
+            min_tokens_per_module=self.min_tokens_per_module,
+        )
+
+    def _preprocess(self):
+        if self.model_args.tie_word_embeddings is True:
+            logger.debug(
+                "The tie_word_embeddings flag is by default set to False. "
+                "This guarantees that the one-shot algorithm saves the final "
+                "weights without errors. Detected tie_word_embeddings=True. "
+                "This may cause issues with the one-shot algorithm on save. "
+            )
+
+        model = self.model_args.model
+        if isinstance(model, str) or isinstance(model, PosixPath):
+            model = initialize_oneshot_model(self.model_args)
+
+        # patch a shared tensor bug in HF transformers
+        # https://github.com/huggingface/transformers/issues/33689
+        patch_tied_tensors_bug(model)
+
+        # wrap model.save_pretrained in compressed_tensors format for vllm
+        modify_save_pretrained(model)
+
+        self.model_args.model = model
+
+        processor = self.model_args.processor
+        if isinstance(processor, str) or processor is None:
+            self.model_args.processor = initialize_processor_from_path(
+                self.model_args, model
+            )
+
+        if self.data_args is not None:
+            self.min_tokens_per_module = self.data_args.min_tokens_per_module
+
+
+def parse_oneshot_args(**kwargs):
+    parser = HfArgumentParser(
+        (OneshotModelArguments, DataTrainingArguments, RecipeArguments)
+    )
+    if not kwargs:
+        model_args, data_args, recipe_args = parser.parse_args_into_dataclasses()
+    else:
+        model_args, data_args, recipe_args = parser.parse_dict(kwargs)
+
+    if recipe_args.recipe_args is not None:
+        if not isinstance(recipe_args.recipe_args, dict):
+            arg_dict = {}
+            for recipe_arg in recipe_args.recipe_args:
+                key, value = recipe_arg.split("=")
+                arg_dict[key] = value
+            recipe_args.recipe_args = arg_dict
+
+    if model_args.tokenizer:
+        if model_args.processor:
+            raise ValueError("Cannot use both a tokenizer and processor")
+
+        logger.debug("Overwriting processor with tokenizer")
+        model_args.processor = model_args.tokenizer
+
+    return model_args, data_args, recipe_args
+
+
+def initialize_oneshot_model(
```

Comment:
redundant?

Reply:
yes thank you. Duplicate code will be removed / combined with existing code